### PR TITLE
feat(llm-bridge): add ollama provider with mistral model support

### PR DESCRIPTION
**Pull request by wujunzhuo**
# Description

By using the [Ollama](https://ollama.com) backend provider and MistralAI models, users could build up the function calling system with all open-source deps from zero.

### CODE REVIEW

File: pkg/bridge/ai/provider/ollama/provider.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+const (
+	defaultSystem = "You are a very helpful assistant. Your job is to choose the best possible action to solve the user question or task."
+
+	systemToolExtra = "If the question of the user matched the description of a tool, the tool will be called, and only the function description JSON object should be returned. Don't make assumptions about what values to plug into functions. Ask for clarification if a user request is ambiguous."
+
+	mistralTmpl = "{{ if .Tools }}[AVAILABLE_TOOLS] {{.Tools}} [/AVAILABLE_TOOLS] {{ end }}[INST] {{ if .System }}{{ .System }} {{ end }}{{ .Prompt }} [/INST]"
+)
+
+func makeOllamaRequestBody(req openai.ChatCompletionRequest) (io.Reader, error) {
+	if !strings.HasPrefix(req.Model, "mistral") {
```
Comment (venjiang) [inline pkg/bridge/ai/provider/ollama/provider.go:61]: There should be default model name: `mistral`, `/invoke` API will not work.

**Review by venjiang — CHANGES_REQUESTED** [review]

**Comment by venjiang** [comment]
Usage needs to be implemented, used by `prompt_eval_count` and `eval_count`, see [ollama API](https://github.com/ollama/ollama/blob/main/docs/api.md)

**Review by venjiang — APPROVED** [review]
👍

**Review by fanweixiao — APPROVED** [review]
🥇

**Review by woorui — APPROVED** [review]

**Pull request merged**